packages/remirror__extension-list/src/list-commands.ts

```diff
@@ -150,21 +147,30 @@ export function splitListItem(listItemType: string | NodeType): CommandFunction
       return true;
     }
 
-    const nextType = $to.pos === $from.end() ? grandParent.contentMatchAt(0).defaultType : null;
-    tr.delete($from.pos, $to.pos);
+    const listItemAttributes = Object.fromEntries(
+      Object.entries(grandParent.attrs).filter(([attr]) => !ignoreAttrs.includes(attr)),
+    );
 
-    const types: TypesAfter = [undefined];
+    // The content inside the list item (e.g. paragraph)
+    const contentType = $to.pos === $from.end() ? grandParent.contentMatchAt(0).defaultType : null;
+    const contentAttributes = { ...$from.node().attrs };
 
-    if (nextType) {
-      types.push({ type: nextType, attrs: {} });
-    }
+    tr.delete($from.pos, $to.pos);
+
+    const types: TypesAfter = contentType
+      ? [
+          { type: listItemType, attrs: listItemAttributes },
+          { type: contentType, attrs: contentAttributes },
+        ]
+      : [{ type: listItemType, attrs: listItemAttributes }];
 
-    if (!canSplit(tr.doc, $from.pos, 2, types)) {
+    if (!canSplit(tr.doc, $from.pos, 2)) {
+      // I can't use `canSplit(tr.doc, $from.pos, 2, types)` and I don't know why
       return false;
     }
 
     if (dispatch) {
-      // TODO: types for tr.split need to be fixed
+      // TODO: types for `tr.split` need to be fixed in `@types/prosemirror-transform`
       dispatch(tr.split($from.pos, 2, types as any).scrollIntoView());
```

Comment:
```suggestion
      // @ts-expect-error
      dispatch(tr.split($from.pos, 2, types).scrollIntoView());
```

This will make sure that we change the any once the types for `prosemirror-transform` are updated.

Reply:
That's cool!